### DISCUSSION TITLE
Feature request: Add support for ONVIF motion detection

### DISCUSSION THREAD

**Discussion by jimsmt** [Sunoo/homebridge-camera-ffmpeg#1196] [Ideas]
@Sunoo Thanks for making this great homebridge plugin, I'm wondering if you can add support for ONVIF, which has motion detection integrated.

I tried the ONVIF intergration on Home Assistant with my Dahua dome camera, two motion sensors and video stream were added to HA and Homekit  automatically. The experience is very good and smooth, once there's any movement, my iPhone will get notification in just 2 seconds with snapshot image.

I noticed that there's already a nodejs library for ONVIF on github, [futomi/node-onvif](https://github.com/futomi/node-onvif), you may take a look

Thank you again

**Comment by Sunoo** [comment]
I’ve actually been thinking about ONVIF recently. Right now, in my mind it would be a helper plugin to handle motion (like the ones I have for FTP and SMTP already), and possibly a utility to search for ONVIF cameras and write out config snippets.

I don’t actually own any ONVIF cameras though, so I’d love to hear opinions from folks who would actually be using it.

**Reply by jimsmt** [reply]
thank you for the quick reply, I'm happy to test if you need

**Comment by jimsmt** [comment]
As a temporary workaround, currently I'm using Home Assistant's automation script to send request to this plugin's HTTP API to trigger motion action, as described [here](https://sunoo.github.io/homebridge-camera-ffmpeg/automation/http.html). It works well

**Reply by effpi** [reply]
Hello @gitgayhub,
could expand how you managed to do this?
Thank you in advance

**Comment by mihailescu2m** [comment]
There's a docker image with onvif2mqqt, which makes a subscription to the motion events from an onvif camera and publishes them to a mqtt server, then you just need to point homebridge-camera-ffmpeg to the mqtt topics.

Confirmed working nicely with my Tapo C200 cameras. Eagerly waiting for HKSV now :)
My config:

```
                {
                    "name": "Tapo-1",
                    "manufacturer": "Tp-Link",
                    "model": "C200",
                    "serialNumber": "2001",
                    "unbridge": true,
                    "motion": true,
                    "motionTimeout": 0,
                    "videoConfig": {
                        "source": "-rtsp_transport tcp -re -i rtsp://user:pass@192.168.0.15:554/stream2",
                        "stillImageSource": "-i rtsp://user:pass@192.168.0.15:554/stream2",
                        "maxWidth": 640,
                        "maxHeight": 480,
                        "maxFPS": 15,
                        "maxBitrate": 384,
                        "forceMax": true,
                        "vcodec": "copy",
                        "audio": true
                    },
                    "mqtt": {
                        "motionTopic": "onvif2mqtt/Tapo-1/motion",
                        "motionMessage": "ON",
                        "motionResetTopic": "onvif2mqtt/Tapo-1/motion",
                        "motionResetMessage": "OFF"
                    }
                },
```

As a side, I am still not 100% happy with the lag. I have Synology Surveillance Station connected to the cameras as well, and with the Tapo app and the Synology app I get ms-range lag (near realtime) on the 1080p feed, while with homebridge-camera-ffmpeg I get ~3s delay on the feed with 640x480 resolution, with an initial +1s if local, or ~+5s if remote extra delay (so when remote I get ~8s until I get the feed to show, which is then delayed ~3s from realtime). Using the 1080p stream adds extra 2-3s :(

Anyone has any suggestion on how I can reduce this lag? You can see I am using unbridge, 640x480 resolution with the copy codec to remove any transcoding.

![https://imgur.com/a/JlhmgET](https://i.imgur.com/hn9swKu.png)
